measures/GEBAppliancesPeakPeriodShift/measure.rb

```diff
@@ -0,0 +1,230 @@
+# frozen_string_literal: true
+
+# see the URL below for information on how to write OpenStudio measures
+# http://nrel.github.io/OpenStudio-user-documentation/reference/measure_writing_guide/
+
+require 'csv'
+
+# start the measure
+class GEBAppliancesPeakPeriodShift < OpenStudio::Measure::ModelMeasure
+  # human readable name
+  def name
+    # Measure name should be the title case of the class name.
+    return 'GEBAppliancesPeakPeriodShift'
+  end
+
+  # human readable description
+  def description
+    return 'TODO'
+  end
+
+  # human readable description of modeling approach
+  def modeler_description
+    return 'TODO'
+  end
+
+  # define the arguments that the user will input
+  def arguments(_model)
+    args = OpenStudio::Measure::OSArgumentVector.new
+
+    arg = OpenStudio::Measure::OSArgument.makeStringArgument('schedules_peak_period', true)
+    arg.setDisplayName('Schedules: Peak Period')
+    arg.setDescription('Specifies the peak period. Enter a time like "15 - 18" (start hour can be 0 through 23 and end hour can be 1 through 24).')
+    arg.setDefaultValue('15 - 18')
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument.makeIntegerArgument('schedules_peak_period_delay', true)
+    arg.setDisplayName('Schedules: Peak Period Delay')
+    arg.setUnits('hr')
+    arg.setDescription('The number of hours after peak period end.')
+    arg.setDefaultValue(0)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument.makeStringArgument('schedules_peak_period_column_names', false)
+    arg.setDisplayName('Schedules: Peak Period Schedule File Column Names')
+    arg.setDescription('Comma-separated list of schedule file column names to shift during the peak period.')
+    args << arg
```

Comment:
Should really clarify that this is specific to ScheduleFile objects.

Do you plan to support another argument similar to this where someone can enter a list of ScheduleRuleset names to shift? That is the most common schedule class used in OpenStudio after all, not that many users have schedules using ScheduleFile.

Reply:
Definitely really early for a review. I'm more testing the overall mechanics here. But yes, my plan is to get the basic functionality in for supporting residential/resstock needs, and then move onto some of the items you are highlighting in these comments...

Reply:
Got it. Yep, I wasn't really reviewing the code, mostly just putting down some thoughts prompted by this PR.